instrumentation/rack/lib/opentelemetry/instrumentation/rack/middlewares/tracer_middleware.rb

```diff
@@ -152,7 +152,7 @@ def create_request_span_name(request_uri_or_path_info, env)
           end
 
           def set_attributes_after_request(span, status, headers, _response)
-            span.status = OpenTelemetry::Trace::Status.error unless (100..499).cover?(status.to_i)
+            span.status = OpenTelemetry::Trace::Status.error if (500..599).cover?(status.to_i)
```

Comment:
based on your comment here https://github.com/open-telemetry/opentelemetry-ruby-contrib/pull/1196/files#r1811615817

Are you saying the `_response` object here is _not_ a `Rack::Response`?

Reply:
Exactly, it is only the raw response array that is transmitted to middleware `[status, headers, body]`